bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -942,18 +941,23 @@ void PreProcessor::registerAndHandleClientPreProcessReqOnNonPrimary(ClientPrePro
   }
 }
 
-const char *PreProcessor::getPreProcessResultBuffer(uint16_t clientId,
-                                                    ReqId reqSeqNum,
-                                                    uint16_t reqOffsetInBatch) const {
-  // Pre-allocated buffers scheme:
+const char *PreProcessor::getPreProcessResultBuffer(uint16_t clientId, ReqId reqSeqNum, uint16_t reqOffsetInBatch) {
+  // Allocate on first use buffers scheme:
   // |first client's first buffer|...|first client's last buffer|......
   // |last client's first buffer|...|last client's last buffer|
   // First client id starts after the last replica id.
   // First buffer offset = numOfReplicas_ * batchSize_
   // The number of buffers per client comes from the configuration parameter clientBatchingMaxMsgsNbr.
   const auto bufferOffset =
       (clientId - numOfReplicas_ - numOfInternalClients_) * clientMaxBatchSize_ + reqOffsetInBatch;
-  LOG_TRACE(logger(), KVLOG(clientId, reqSeqNum, reqOffsetInBatch, bufferOffset));
+  LOG_INFO(logger(), KVLOG(clientId, reqSeqNum, reqOffsetInBatch, bufferOffset));
+  if (preProcessResultBuffers_.find(bufferOffset) == preProcessResultBuffers_.end()) {
+    LOG_INFO(logger(), "Allocating buffer in offset: " << bufferOffset);
+    preProcessResultBuffers_[bufferOffset] = Sliver(new char[maxPreExecResultSize_], maxPreExecResultSize_);
+    LOG_INFO(logger(), "Allocate complete");
```

Comment:
Do we really need this and the next log info message? This looks more like a debug message to me.

Reply:
Was added for debugging, deleted all unnecessary logs.